src/Provider/3dTilesProvider.js

```diff
@@ -91,23 +96,36 @@ function executeCommand(command) {
     };
     if (path) {
         // Check if we have relative or absolute url (with tileset's lopocs for example)
-        const url = path.startsWith('http') ? path : metadata.baseURL + path;
+        let url = path.startsWith('http') ? path : metadata.baseURL + path;
+        if (layer.source.isC3DTilesGoogleSource) {
+            url = layer.source.getTileUrl(url);
+        }
         const supportedFormats = {
             b3dm: b3dmToMesh,
             pnts: pntsParse,
+            gltf: gltfToMesh,
         };
         return Fetcher.arrayBuffer(url, layer.source.networkOptions).then((result) => {
             if (result !== undefined) {
                 let func;
                 const magic = utf8Decoder.decode(new Uint8Array(result, 0, 4));
                 if (magic[0] === '{') {
                     result = JSON.parse(utf8Decoder.decode(new Uint8Array(result)));
-                    const newPrefix = url.slice(0, url.lastIndexOf('/') + 1);
+                    let newPrefix = '';
```

Comment:
No need to assign `newPrefix` since it is assigned in both branch of the following condition (or do a ternary condition).
```suggestion
                    let newPrefix;
```